hypothesis-python/src/hypothesis/strategies.py

```diff
@@ -386,6 +398,13 @@ def floats(
     Where not explicitly ruled out by the bounds, all of infinity, -infinity
     and NaN are possible values generated by this strategy.
 
+    The width argument specifies the maximum number of bits of precision
+    required to represent the generated float. Valid values are 16, 32, or 64.
+    Passing ``width=32`` will still use the builtin 64-bit ``float`` class,
+    but always for values which can be exactly represented as a 32-bit float.
+    Half-precision floats (``width=16``) require either Numpy, or Python 3.6+
+    where this type is supported by the :mod:`python:struct` module.
```

Comment:
nit: I'm not sure we need to explain the struct bit in the docstring. It's just an implementation detail. I think it suffices to just say that these are the requirements.